### PR TITLE
Event controller

### PR DESCRIPTION
**Pull request by mangelajo**
The event controller subscribes to the objects it needs to watch
and sends the events to an event handler registry.

This includes a network-plugin-syncer stub as reference.

Resolves-Issue: #857
    
Co-Authored-By: Sridhar Gaddam <sgaddam@redhat.com>
Signed-off-by: Miguel Angel Ajo <majopela@redhat.com>

### CODE REVIEW

**Review by tpantelis — APPROVED** [review]

File: pkg/event/controller/controller_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+			Eventually(testEvents).Should(Receive(Equal(
+				testing.TestEvent{Handler: testHandlerName, Name: testing.EvNodeCreated, Parameter: node})))
+			Consistently(testEvents).ShouldNot(Receive())
+
+			node.Labels = map[string]string{"labeled-i-am": "i-am"}
+
+			test.UpdateResource(nodes, node)
+
+			Eventually(testEvents).Should(Receive(Equal(
```
Comment (sridhargaddam) [inline pkg/event/controller/controller_test.go:92]: @mangelajo cool. So you added a test-case to verify that we are not getting events when labels are updated. Nice.

**Review by sridhargaddam — APPROVED** [review]
LGTM

File: pkg/event/controller/controller_test.go (hunk 1 above)
Comment (mangelajo) [reply pkg/event/controller/controller_test.go:92]: I actually removed that filter for now. I will be needing labels and annotations for OVN.

We can later add some interface from the controller consumers to let us install specific filters. I hope that sounds ok @sridhargaddam

Comment (sridhargaddam) [reply pkg/event/controller/controller_test.go:92]: Okay @mangelajo

**Pull request merged**